lib/index.js

```diff
@@ -84,12 +84,33 @@ var lookup = [
  * @returns {number}
  */
 var generate = module.exports = function generate(str) {
-  var i = 1;
+  var i = 0;
   var len = str.length;
-  var result = lookup[str.charCodeAt(0) & 0xFF];
+  var start = -1;
+  var result = 0;
+  var char;
 
   while (i < len) {
-    result = lookup[(str.charCodeAt(i++) ^ (result >> 8)) & 0xFF] ^ (result << 8);
+    // Read all characters and safe them in a intermediate array to prevent rereading them
```

Comment:
I removed that, improved the perf on your PR, pushing, about 100k more ops/s on hashes, non hashes roughly the same still.